aten/src/ATen/AccumulateType.h

```diff
@@ -35,6 +35,8 @@ template <> struct AccumulateType<int32_t, true> { using type = int64_t; };
 template <> struct AccumulateType<int64_t, true> { using type = int64_t; };
 template <> struct AccumulateType<bool, true> {using type = bool; };
 template <> struct AccumulateType<BFloat16, false> { using type = float; };
+template <> struct AccumulateType<std::complex<float>, false> { using type = std::complex<double>; };
```

Comment:
@anjali411 Is `std::complex<float>` not enough for accumulating? 

Currently on master for `c10::complex<float>` , on CPU we accumulate to `c10::complex<double>` while on CUDA we accumulate to `c10::complex<float>` itself.
https://github.com/pytorch/pytorch/blob/79e5b053b690852b21d881357904bc5a4438d95b/aten/src/ATen/AccumulateType.h#L71-L74

Reply:
I see that it is similar to `float` on the line below

```
template <> struct AccumulateType<float, false> { using type = double; };
```

Reply:
on CUDA double precision operations are expensive and so by default, we don't actually want to use double as an acc_type on CUDA.